### ISSUE TITLE
support configuring allocatable resource of member clusters

### ISSUE THREAD

**Issue by merryzhou** [karmada-io/karmada#1634]
**What would you like to be added**:

Suppose a member cluster with total resource: 1000 CPU, and 200 CPU can be used by applications that deployed through karmada 

**Why is this needed**:

my member cluster can be used directly, and I want to guarantee the total amount of available resources of the member cluster

**Comment by RainbowMango** [comment]
> Suppose a member cluster with total resource: 1000 CPU, and 200 CPU can be used by applications that deployed through karmada

Do you mean you want to restrict the resource quota for applications that are deployed by Karmada?  These applications can't exceed a hard limit(200CPU in your case)?

**Comment by merryzhou** [comment]
> These applications can't exceed a hard limit(200CPU in your case)?

yes

**Comment by RainbowMango** [comment]
Then, I guess the [FederatedResourceQuota](https://github.com/karmada-io/karmada/blob/bb46766f63b52a0fea41c84b9fc2f42aeb1409a1/pkg/apis/policy/v1alpha1/federatedresourcequota_types.go#L25) might help. You can set the quota for each namespaces that Karmada might sync workload to properly. 
The scheduler will prevent workload from scheduling to that cluster if the quota exceeds the limit(The implementation hasn't done yet.)

I'm interested in your use case. Can you take a look the `FederatedResourceQuota`? Looking forward to your feedback.

**Comment by merryzhou** [comment]
FederatedResourceQuota is to restrict resources in namespace level， I want to restrict resources in cluster level.

suppose a member cluster has 1000 CPU, and the restrict is 200 CPU. What I want is the total resources of all applications that deployed by karmada to the member cluster is not exceed 200 CPU.  

In this case, I don't care about the ns count nor the ns resourceQuota.

**Comment by RainbowMango** [comment]
Yes, you have to manage the total namespace quota across all clusters.

It's essentially the quota thing. Maybe we can introduce another quota, like ClusterQuota?

**Comment by merryzhou** [comment]
clusterQuota might be a solution

**Comment by RainbowMango** [comment]
I'm not sure if this is a common use case, but I'm glad if you can lead the effort to do it.

**Comment by merryzhou** [comment]
ok，I'll submit a pr for it ASAP.

**Comment by RainbowMango** [comment]
A PR for what? proposal or feature implementation? I'm ok with the proposal, the implementation thing should be the next thing after discussion.

**Comment by merryzhou** [comment]
> A PR for what? proposal or feature implementation? I'm ok with the proposal, the implementation thing should be the next thing after discussion.

I mean a proposal pr

**Comment by RainbowMango** [comment]
Thanks. 
/assign @merryzhou

**Comment by XiShanYongYe-Chang** [comment]
Hi @merryzhou, are you still working on this issue?
